lib/ProductOpener/Tags.pm

```diff
@@ -1953,9 +1954,9 @@ sub display_tag_link($$) {
 	}
 
 	if ($tagtype eq 'emb_codes') {
-
 		my $city_code = get_city_code($tagid);
 
+		init_emb_codes();
```

Comment:
I think we should remove the init_emb_codes() calls inside functions that may need %emb_codes, and assume it has been loaded by startup.pl our the scripts like update_all_products.pl etc. that need it. That will remove a huge number of function calls.

We could also initialize %emb_codes to an empty hash instead of being undefined, that way the functions would not die with errors if they are called without init_emb_codes() having been called.

Reply:
> 
> 
> I think we should remove the init_emb_codes() calls inside functions that may need %emb_codes, and assume it has been loaded by startup.pl our the scripts like update_all_products.pl etc. that need it. That will remove a huge number of function calls.

I agree with you.

> We could also initialize %emb_codes to an empty hash instead of being undefined, that way the functions would not die with errors if they are called without init_emb_codes() having been called.

I would prefer if it threw an error. That's because we would know there is something wrong, instead of just having no `emb_codes`.

Reply:
It's a few API calls, but (according to my test logs) not too many, since the sub is only called s few times. I'll revisit the PR later this week.

Reply:
I changed my mind. :) It's actually breaking a lot of scripts (all the ones that update products in some way). I'll replace the die statements by a call to init_emb_codes() if it's not defined.